app/models/atmosphere/appliance.rb

```diff
@@ -163,6 +163,12 @@ def assign_fund
         vmt.tenants.map(&:funds)
       end.flatten.uniq.compact
 
+      # Restrict to funds which belong to this appliance's owner
+      funds = funds & appliance_set.user.funds
+
+      # Restrict to funds which match this appliance's set of selected tenants
+      funds = funds & tenants.map(&:funds) if tenants.present?
```

Comment:
`funds` is not an AR relation. Sorry, it's late. If you can rewrite the query in lines 162-164 in pure arel (I tried some stuff but am unable to reproduce the required behavior)  then I'll introduce this change; otherwise I believe any potential performance gain is absolutely trivial and not worth the effort.